source/server/config/network/http_connection_manager.cc

```diff
@@ -85,6 +85,36 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(const Json::Object& con
     use_remote_address_ = config.getBoolean("use_remote_address");
   }
 
+  if (config.hasObject("forward_client_cert")) {
+    // TODO: Check mTLS.
+    const std::string forward_client_cert = config.getString("forward_client_cert");
+    if (forward_client_cert == "forward_only") {
+      forward_client_cert_ = Http::ForwardClientCertType::ForwardOnly;
+    } else if (forward_client_cert == "append_forward") {
+      forward_client_cert_ = Http::ForwardClientCertType::AppendForward;
+    } else if (forward_client_cert == "sanitize_set") {
+      forward_client_cert_ = Http::ForwardClientCertType::SanitizeSet;
+    } else if (forward_client_cert == "sanitize") {
+      forward_client_cert_ = Http::ForwardClientCertType::Sanitize;
+    } else {
+      ASSERT(forward_client_cert == "always_forward_only");
+      forward_client_cert_ = Http::ForwardClientCertType::AlwaysForwardOnly;
+    }
+  } else {
+    forward_client_cert_ = Http::ForwardClientCertType::Sanitize;
+  }
+
+  if (config.hasObject("set_client_cert_details")) {
```

Comment:
BTW, can you contribute the new config stuff to https://github.com/lyft/envoy-api as well? Thanks.